plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php

```diff
@@ -606,5 +428,101 @@ private function register_post_type_for_order_placeholders(): void {
 		);
 	}
 
+	/**
+	 * Returns the HPOS setting for rendering in Features section of the settings page.
+	 *
+	 * @param array  $feature_setting HPOS feature value as defined in the feature controller.
+	 * @param string $feature_id ID of the feature.
+	 *
+	 * @return array Feature setting object.
+	 */
+	private function get_hpos_feature_setting( array $feature_setting, string $feature_id ) {
+		if ( ! in_array( $feature_id, array( self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION, DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION, 'custom_order_tables' ), true ) ) {
+			return $feature_setting;
+		}
+
+		if ( Constants::get_constant( 'WC_INSTALLING', false ) ) {
+			return $feature_setting;
+		}
+
+		$sync_status = $this->data_synchronizer->get_sync_status();
+		switch ( $feature_id ) {
+			case self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION:
+				return $this->get_hpos_setting_for_feature( $sync_status );
+			case DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION:
+				return $this->get_hpos_setting_for_sync( $sync_status );
+			case 'custom_order_tables':
+				return array();
+		}
+	}
 
+	/**
+	 * Returns the HPOS setting for rendering HPOS vs Post setting block in Features section of the settings page.
+	 *
+	 * @param array $sync_status Details of sync status, includes pending count, and count when sync started.
+	 *
+	 * @return array Feature setting object.
+	 */
+	private function get_hpos_setting_for_feature( $sync_status ) {
+		$hpos_enabled            = $this->custom_orders_table_usage_is_enabled();
+		$plugin_info             = $this->features_controller->get_compatible_plugins_for_feature( 'custom_order_tables', true );
+		$plugin_incompat_warning = $this->plugin_util->generate_incompatible_plugin_feature_warning( 'custom_order_tables', $plugin_info );
+		$sync_complete           = 0 === $sync_status['current_pending_count'];
+		$disabled_option         = array();
+		if ( count( array_merge( $plugin_info['uncertain'], $plugin_info['incompatible'] ) ) > 0 ) {
+			$disabled_option = array( 'yes' );
+		}
+		if ( ! $sync_complete ) {
+			$disabled_option = array( 'yes', 'no' );
+		}
+
+		return array(
+			'id'          => self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION,
+			'title'       => __( 'Data storage for orders', 'woocommerce' ),
+			'type'        => 'radio',
+			'options'     => array(
+				'no'  => __( 'WordPress Post tables', 'woocommerce' ),
```

Comment:
This is a nit, but I don't think `Post` should be capitalized (and it doesn't tend to be in other places where it's used in a sentence or phrase within the WP UI):

```suggestion
				'no'  => __( 'WordPress post tables', 'woocommerce' ),
```